config/configmapprovider/provider.go

```diff
@@ -79,9 +76,12 @@ type Retrieved interface {
 	//
 	// This method must be called when the service ends, either in case of success or error.
 	//
+	// This method must not call the `onChange` callback and must ensure that the callback is
```

Comment:
This change is not clear to me. Previously `Close` could be used to interrupt a pending watch. As a result of calling `Close` we could end up calling `onChance` callback (not necessarily directly called by `Close`, but initiated by `Close`).
If we remove this line how do we terminate a pending watch? Is it still done via Close? The only difference is that watcher is not going to now that the watch is terminated? I am not sure how this would work. Does it become the responsibility of the caller of `Close` to let the watcher know that they should no longer wait for `onChange` to be called?

Reply:
Close stops the "provider" watcher if any, and guarantees that the onChange will NOT be called after the method returns. The difference from the previous text is that I don't need the "provider" to call "onChange" with a "ErrSessionClosed" error, it does not give me any information since the implementation of "onChange" also calls Close and waits for that to return.

The return of the "Close" func is equivalent with the guarantee that the provider is Closed, all its gorutines are done, and will never call onChange.

Reply:
Done.